src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinderProvider.cs

```diff
@@ -20,9 +19,7 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (context.Metadata.IsComplexType &&
-                !context.Metadata.IsCollectionType &&
-                HasDefaultConstructor(context.Metadata.ModelType.GetTypeInfo()))
+            if (context.Metadata.IsComplexType && !context.Metadata.IsCollectionType)
```

Comment:
Sorry missed this comment earlier. What's the old behavior we're striving for here? The purpose of the patch is to revert breaking changes, but without changing any other behavior.